thrust/thrust/detail/reference.h

```diff
@@ -161,7 +161,7 @@ class reference
    *
    *  \param other The \p tagged_reference to swap with.
    */
-  _CCCL_HOST_DEVICE void swap(derived_type& other)
+  _CCCL_HOST_DEVICE void swap(derived_type other)
```

Comment:
That feels like a bug

Reply:
I modelled this after `swap` for `vector<bool>::reference`, see: https://en.cppreference.com/w/cpp/container/vector_bool/swap

How should I implement swap for proxy references otherwise? The alternative are two swap overloads, one for l-value and one for r-value, or a constrained overload with a forwarding reference.

Reply:
yeah totally forgot that we are returning a proxy